lib/importers/user_importer.rb

```diff
@@ -75,6 +75,30 @@ def self.update_users(users=nil, wiki=nil)
     end
   end
 
+  # There are some users who updated their username through MetaWiki, and
+  # to get their updated username get reflected in the dahsboard, they need to relogin.
+  # So, this method will update the username without making the user relogin
+  def self.update_username_for_changed_metawiki_usernames(username)
```

Comment:
Don't duplicate all this code for sanitizing input. Extract it to a new method, and then call that method from `new_from_username`.

Reply:
Oh, ya sorry I'll do it